In [2]:
import pandas as pd

In [12]:
candidatas_pe_2020 = pd.read_csv('base_dados/consulta_cand_2020_PE.csv', sep=';', encoding='latin_1')

In [13]:
candidatas_pe_2020.head()

DT_GERACAO HH_GERACAO  ANO_ELEICAO  CD_TIPO_ELEICAO    NM_TIPO_ELEICAO  \
0  05/08/2021   12:18:13         2020                2  ELEIÇÃO ORDINÁRIA   
1  05/08/2021   12:18:13         2020                2  ELEIÇÃO ORDINÁRIA   
2  05/08/2021   12:18:13         2020                2  ELEIÇÃO ORDINÁRIA   
3  05/08/2021   12:18:13         2020                2  ELEIÇÃO ORDINÁRIA   
4  05/08/2021   12:18:13         2020                2  ELEIÇÃO ORDINÁRIA   

   NR_TURNO  CD_ELEICAO                DS_ELEICAO  DT_ELEICAO TP_ABRANGENCIA  \
0         1         426  Eleições Municipais 2020  15/11/2020      MUNICIPAL   
1         1         426  Eleições Municipais 2020  15/11/2020      MUNICIPAL   
2         1         426  Eleições Municipais 2020  15/11/2020      MUNICIPAL   
3         1         426  Eleições Municipais 2020  15/11/2020      MUNICIPAL   
4         1         426  Eleições Municipais 2020  15/11/2020      MUNICIPAL   

   ... DS_SIT_TOT_TURNO  ST_REELEICAO ST_DECLARAR_BENS  \
0  ...         SUPLENTE             N                N   
1  ...         SUPLENTE             N                S   
2  ...         SUPLENTE             N                S   
3  ...    ELEITO POR QP             N                S   
4  ...           #NULO#             N                S   

   NR_PROTOCOLO_CANDIDATURA          NR_PROCESSO  \
0                        -1  6004724720206170105   
1                        -1  6000665220206170064   
2                        -1  6001433520206170008   
3                        -1  6001353020206170082   
4                        -1  6002360820206170037   

   CD_SITUACAO_CANDIDATO_PLEITO  DS_SITUACAO_CANDIDATO_PLEITO  \
0                             2                      DEFERIDO   
1                             2                      DEFERIDO   
2                             2                      DEFERIDO   
3                             2                      DEFERIDO   
4                            -1                        #NULO#   

  CD_SITUACAO_CANDIDATO_URNA DS_SITUACAO_CANDIDATO_URNA  \
0                          2                   DEFERIDO   
1                          2                   DEFERIDO   
2                          2                   DEFERIDO   
3                          2                   DEFERIDO   
4                         -1                     #NULO#   

  ST_CANDIDATO_INSERIDO_URNA  
0                        SIM  
1                        SIM  
2                        SIM  
3                        SIM  
4                        NÃO  

[5 rows x 63 columns]

In [14]:
candidatas_pe_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21101 entries, 0 to 21100
Data columns (total 63 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   DT_GERACAO                     21101 non-null  object 
 1   HH_GERACAO                     21101 non-null  object 
 2   ANO_ELEICAO                    21101 non-null  int64  
 3   CD_TIPO_ELEICAO                21101 non-null  int64  
 4   NM_TIPO_ELEICAO                21101 non-null  object 
 5   NR_TURNO                       21101 non-null  int64  
 6   CD_ELEICAO                     21101 non-null  int64  
 7   DS_ELEICAO                     21101 non-null  object 
 8   DT_ELEICAO                     21101 non-null  object 
 9   TP_ABRANGENCIA                 21101 non-null  object 
 10  SG_UF                          21101 non-null  object 
 11  SG_UE                          21101 non-null  int64  
 12  NM_UE                          21101 non-null 

In [15]:
candidatas_pe_2020.columns

Index(['DT_GERACAO', 'HH_GERACAO', 'ANO_ELEICAO', 'CD_TIPO_ELEICAO',
       'NM_TIPO_ELEICAO', 'NR_TURNO', 'CD_ELEICAO', 'DS_ELEICAO', 'DT_ELEICAO',
       'TP_ABRANGENCIA', 'SG_UF', 'SG_UE', 'NM_UE', 'CD_CARGO', 'DS_CARGO',
       'SQ_CANDIDATO', 'NR_CANDIDATO', 'NM_CANDIDATO', 'NM_URNA_CANDIDATO',
       'NM_SOCIAL_CANDIDATO', 'NR_CPF_CANDIDATO', 'NM_EMAIL',
       'CD_SITUACAO_CANDIDATURA', 'DS_SITUACAO_CANDIDATURA',
       'CD_DETALHE_SITUACAO_CAND', 'DS_DETALHE_SITUACAO_CAND', 'TP_AGREMIACAO',
       'NR_PARTIDO', 'SG_PARTIDO', 'NM_PARTIDO', 'SQ_COLIGACAO',
       'NM_COLIGACAO', 'DS_COMPOSICAO_COLIGACAO', 'CD_NACIONALIDADE',
       'DS_NACIONALIDADE', 'SG_UF_NASCIMENTO', 'CD_MUNICIPIO_NASCIMENTO',
       'NM_MUNICIPIO_NASCIMENTO', 'DT_NASCIMENTO', 'NR_IDADE_DATA_POSSE',
       'NR_TITULO_ELEITORAL_CANDIDATO', 'CD_GENERO', 'DS_GENERO',
       'CD_GRAU_INSTRUCAO', 'DS_GRAU_INSTRUCAO', 'CD_ESTADO_CIVIL',
       'DS_ESTADO_CIVIL', 'CD_COR_RACA', 'DS_COR_RACA', 'CD_OCUPACAO',
       '

In [16]:
colunas_categoricas = ['NM_TIPO_ELEICAO','DS_ELEICAO','TP_ABRANGENCIA','SG_UF', 'NM_UE', 'DS_CARGO', 'NM_CANDIDATO', 'NM_URNA_CANDIDATO', 'NM_SOCIAL_CANDIDATO',
                       'NM_EMAIL', 'DS_SITUACAO_CANDIDATURA', 'TP_AGREMIACAO', 'SG_PARTIDO', 'NM_PARTIDO', 'NM_COLIGACAO', 'DS_COMPOSICAO_COLIGACAO',
                       'DS_NACIONALIDADE', 'SG_UF_NASCIMENTO','NM_MUNICIPIO_NASCIMENTO', 'DS_GENERO', 'DS_GRAU_INSTRUCAO', 'DS_ESTADO_CIVIL', 'DS_COR_RACA',
                       'DS_OCUPACAO', 'VR_DESPESA_MAX_CAMPANHA', 'DS_SIT_TOT_TURNO', 'ST_REELEICAO', 'ST_DECLARAR_BENS', 'DS_SITUACAO_CANDIDATO_PLEITO', 
                       'DS_SITUACAO_CANDIDATO_URNA','ST_CANDIDATO_INSERIDO_URNA'] 

Localize nas colunas_categoricas do dataframe candidatas_pe_2020 todas as linhas que possuem valores #NULO# e susbtitua por None:

In [17]:
candidatas_pe_2020[colunas_categoricas] = candidatas_pe_2020.loc[:, colunas_categoricas].replace({'#NULO#':None})

In [18]:
candidatas_pe_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21101 entries, 0 to 21100
Data columns (total 63 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   DT_GERACAO                     21101 non-null  object 
 1   HH_GERACAO                     21101 non-null  object 
 2   ANO_ELEICAO                    21101 non-null  int64  
 3   CD_TIPO_ELEICAO                21101 non-null  int64  
 4   NM_TIPO_ELEICAO                21101 non-null  object 
 5   NR_TURNO                       21101 non-null  int64  
 6   CD_ELEICAO                     21101 non-null  int64  
 7   DS_ELEICAO                     21101 non-null  object 
 8   DT_ELEICAO                     21101 non-null  object 
 9   TP_ABRANGENCIA                 21101 non-null  object 
 10  SG_UF                          21101 non-null  object 
 11  SG_UE                          21101 non-null  int64  
 12  NM_UE                          21101 non-null 

Na linha 19 apenas 7 valores não são nulos. Antes de fazer o replace, o pandas informava que nessa linha 21101 valores eram não nulos. Um difeça também é notável na linha 59. De 21101 passou para 20431 valores não nulos. 